utils/updateProvenanceHash.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const crypto = require('crypto');
+const isLocal = typeof process.pkg === "undefined";
+const basePath = isLocal ? process.cwd() : path.dirname(process.execPath);
+
+const { baseUri } = require("../src/config.js");
+var provenanceHashList = [];
+
+const generatProvenance = (_provenanceHashList) => {
```

Comment:
(Apologies if I'm overstepping any bounds, maintainers! Don't see any docs about code reviewing, and I just took a look through some of the open PRs)

Small typo

```suggestion
const generateProvenance = (_provenanceHashList) => {
```

Also this and the `generateProvenance` below this are pretty similar, I think it'd be worth combining the functionality.